hoki/load.py

```diff
@@ -410,11 +411,173 @@ def _UV_nebular_lines(path):
 
     return pd.read_csv(path, skiprows=1, sep=r'\s+', engine='python', names=column_UV_em_lines)
 
+
+
+#####################################
+#  BPASS Load over all metallicity  #
+#####################################
+
+def all_rates(data_path, imf, binary=True):
+    """
+    Loads the BPASS supernova event files.
+
+    Notes
+    -----
+    The rates are just read from file and not normalised.
+
+    Input
+    -----
+    data_path : `str`
+        The filepath to the folder containing the BPASS data
+
+    binary : `bool`
+        Use the binary files or just the single stars. Default=True
+
+    imf : `str`
+        BPASS Identifier of the IMF to be used.
+        The accepted IMF identifiers are:
+        - `"imf_chab100"`
+        - `"imf_chab300"`
+        - `"imf100_100"`
+        - `"imf100_300"`
+        - `"imf135_100"`
+        - `"imf135_300"`
+        - `"imfall_300"`
+        - `"imf170_100"`
+        - `"imf170_300"`
+
+    Returns
+    -------
+    `pandas.DataFrame` (51, (8,13))
+        A pandas MultiIndex dataframe containing the BPASS number of events
+        per metallicity per type.
+        Usage: rates.loc[log_age, (type, metallicity)]
+
+
+        Notes
+        -----
+        This dataframe has the following structure.
+        The index is the log_age as a float.
+        The column is a `pandas.MultiIndex` with the event types
+        (level=0, `float`) and the metallicity (level=1, `float`)
+
+        |Event Type | Ia      | IIP      |  ... | PISNe | low_mass |
+        |Metallicity| 0.00001 | 0.00001  |  ... |  0.04 |    0.04  |
+        | log_age   |---------|----------|------|-------|----------|
+        |    6.0    |
+        |    ...    |                  Event Rate values
+        |    11.0   |
+
+    """
+
+    # Check population type
+    star = "bin" if binary else "sin"
+
+    # Check if the given IMF is in the accepted IMFs
+    if imf not in BPASS_IMFS:
+        raise HokiKeyError(
+            f"{imf} is not a BPASS IMF. Please select a correct IMF.\n"\
+            "These can be found in the documentation of this function.")
+
+    # Create the output DataFrame
+    arrays = [BPASS_NUM_METALLICITIES, BPASS_EVENT_TYPES]
+    columns = pd.MultiIndex.from_product(
+        arrays, names=["Metallicicty", "Event Type"])
+
+    rates = pd.DataFrame(index=BPASS_TIME_BINS,
+                         columns=columns,
+                         dtype=np.float64)
+    rates.index.name = "log_age"
+
+    # load supernova count files
+    for num, metallicity in enumerate(BPASS_METALLICITIES):
+        data = model_output(
+            f"{data_path}/supernova-{star}-{imf}.z{metallicity}.dat"
+        )
+        data = data.loc[:, slice(BPASS_EVENT_TYPES[0], BPASS_EVENT_TYPES[-1])]
+
+        rates.loc[:, (BPASS_NUM_METALLICITIES[num], slice(None))] = data.to_numpy()
+
+    # swap metallicity and event type
+    return rates.swaplevel(0, 1, axis=1)
+
+
+def all_spectra(data_path, imf, binary=True):
```

Comment:
this name is too vague - at face value it doesn't tell your user what exactly is being loaded and that a file is being created automatically. we can talk about this

Reply:
`spectra_all_z_cached`?

Reply:
named `spectra_all_z` in commit 7801d16